go/libraries/doltcore/sqle/enginetest/dolt_queries_rebase.go

```diff
@@ -0,0 +1,702 @@
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:          "call dolt_rebase('-i', 'main');",
+				ExpectedErrStr: dprocedures.ErrRebaseUncommittedChanges.Error(),
+			},
+			{
+				Query:    "call dolt_add('t');",
+				Expected: []sql.Row{{0}},
+			},
+			{
+				Query:          "call dolt_rebase('-i', 'main');",
+				ExpectedErrStr: dprocedures.ErrRebaseUncommittedChanges.Error(),
+			},
+		},
+	},
+	{
+		SkipPrepared: true,
+		Name:         "dolt_rebase errors: no database selected",
+		SetUpScript: []string{
+			"create database temp;",
+			"use temp;",
+			"drop database temp;",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:    "select database();",
+				Expected: []sql.Row{{nil}},
+			},
+			{
+				// TODO: This test isn't working because AssertErr is called and currently always
+				//       creates a new Context, which is always initialized with mydb as the current
+				//       database. If we changed evaluation.go:126 to use AssertErrWithCtx instead and
+				//       reused the existing Context instance, then we could probably make this work.
+				Skip:           true,
+				Query:          "call dolt_rebase('-i', 'main');",
+				ExpectedErrStr: "no database selected",
+			},
+		},
+	},
+	{
+		Name: "dolt_rebase errors: active merge, cherry-pick, or rebase",
+		SetUpScript: []string{
+			"create table t (pk int primary key, col1 varchar(100));",
+			"call dolt_commit('-Am', 'creating table t');",
+			"call dolt_branch('branch1');",
+			"insert into t values (0, 'zero');",
+			"call dolt_commit('-am', 'inserting row 0');",
+
+			"call dolt_checkout('branch1');",
+			"insert into t values (0, 'nada');",
+			"call dolt_commit('-am', 'inserting row 0');",
+
+			"set @@autocommit=0;",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				// Merging main creates a conflict, so we're in an active
+				// merge until we resolve.
+				Query:    "call dolt_merge('main');",
+				Expected: []sql.Row{{"", 0, 1}},
+			},
+			{
+				Query:          "call dolt_rebase('-i', 'main');",
+				ExpectedErrStr: "unable to start rebase while a merge is in progress – abort the current merge before proceeding",
+			},
+		},
+	},
+	{
+		Name: "dolt_rebase errors: rebase working branch already exists",
+		SetUpScript: []string{
+			"create table t (pk int primary key);",
+			"call dolt_commit('-Am', 'creating table t');",
+			"call dolt_branch('branch1');",
+			"call dolt_branch('dolt_rebase_branch1');",
+
+			"insert into t values (0);",
+			"call dolt_commit('-am', 'inserting row 0');",
+
+			"call dolt_checkout('branch1');",
+			"insert into t values (1);",
+			"call dolt_commit('-am', 'inserting row 1');",
+			"insert into t values (10);",
+			"call dolt_commit('-am', 'inserting row 10');",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:          "call dolt_rebase('-i', 'main');",
+				ExpectedErrStr: "fatal: A branch named 'dolt_rebase_branch1' already exists.",
```

Comment:
This isn't great, these are supposed to be strictly temporary identifiers to get around us not supporting a detached head operation. Would be better to just choose a branch name at random, put a number on the end, something else. No reason to ever make the customer care about our branch namespace choice here.

Reply:
I'm on the fence about this. It's an easy change to use a different working branch name, but I liked that the pattern was predictable, i.e. "dolt_rebase_" + rebaseBranchName. The main reason I thought a predictable pattern would be useful was if a user needed to reconnect in a different session and would have to check out this working branch to continue the rebase. 

We don't want to allow multiple active rebases to the same branch at the same time, and the "dolt_" branch namespace shouldn't be used by customers, so I don't think the naming restriction causes a problem.